code/python3/search_letor.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python
+#
+# Simple example script demonstrating how to re-rank using the trained model.
+#
+# Copyright (C) 2019 Vaibhav Kansagara
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+# USA
+
+import sys
+import xapian
+import xapianletor
+
+# We require three command line arguments.
+if len(sys.argv) != 4:
+    print("Usage: %s DATABASE MSIZE QUERY" % sys.argv[0],
+          "NB: QUERY should be quoted to protect it from the shell.",
+          sep='\n', file=sys.stderr)
+    sys.exit(1)
+
+try:
+    db_path = sys.argv[1]
+    msize = sys.argv[2]
+    query_string = sys.argv[3]
+
+    db = xapian.Database(db_path)
+
+    parser = xapian.QueryParser()
+    parser.add_prefix("title", "S")
+    parser.add_prefix("subject", "S")
+    parser.set_database(db)
+    parser.set_default_op(xapian.Query.OP_OR)
+    parser.set_stemmer(xapian.Stem("en"))
+    parser.set_stemming_strategy(xapian.QueryParser.STEM_SOME)
+
+    query_no_prefix = parser.parse_query(query_string,
+                                         parser.FLAG_DEFAULT|
+                                         parser.FLAG_SPELLING_CORRECTION)
+    # query with title as default prefix
+    query_default_prefix = parser.parse_query(query_string,
+                                              parser.FLAG_DEFAULT|
+                                              parser.FLAG_SPELLING_CORRECTION,
+                                              "S")
```

Comment:
You don't appear to use spelling correction here, so passing these flags is just makes the example less clear (I guess you started from an example demonstrating spelling correction - if so, you should preserve the copyright statements from that rather than claiming this is all your own work).

Reply:
fixed in a fixup commit.